src/app/GitUI/UserControls/FileStatusDiffCalculator.cs

```diff
@@ -229,9 +229,11 @@ private List<FileStatusWithDescription> CalculateDiffs(
             IReadOnlyList<GitItemStatus> allBaseToA = module.GetDiffFilesWithSubmodulesStatus(baseRevId, firstRev.ObjectId, firstRev.FirstParentId, cancellationToken);
 
             GitItemStatusNameEqualityComparer comparer = new();
-            List<GitItemStatus> sameBaseToAandB = allBaseToB.Intersect(allBaseToA, comparer).Except(allAToB, comparer).ToList();
-            List<GitItemStatus> onlyA = allBaseToA.Except(allBaseToB, comparer).ToList();
-            List<GitItemStatus> onlyB = allBaseToB.Except(allBaseToA, comparer).ToList();
+            List<GitItemStatus> sameBaseToAandB = [.. allBaseToB
+                .Intersect(allBaseToA, comparer)
+                .Except(allAToB.Where(i => !((i.IsRenamed || i.IsCopied) && i.RenameCopyPercentage == "100")), comparer)];
```

Comment:
This subexpression is worth having a name, e.g.
```
GitItemStatus[] allAToBExceptExactRenameCopy = [.. allAToB.Where(i => !((i.IsRenamed || i.IsCopied) && i.RenameCopyPercentage == "100"))];
```
For performance reasons, we should use arrays instead of `List<>`.

Reply:
note also that allAToBExceptExactRenameCopy etc only needs the path so e.g. a HashSet could be used, but that should not make a noticeable difference